tests/DayOfMonthTest.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brick\DateTime\Tests;
+
+use Brick\DateTime\Field\DayOfMonth;
+
+/**
+ * Unit tests for class DayOfMonth.
+ */
+class DayOfMonthTest extends AbstractTestCase
+{
+    public function testCheckWithNullMonthOfYear()
+    {
+        $dayOfMonth = new DayOfMonth();
+
+        $this->assertNull($dayOfMonth->check(31));
+    }
+}
```

Comment:
Rather than testing the `DayOfMonth` field (and testing only one edge case in this class) it would be more interesting to add a test on the calling code (in `LocalDate` or `MonthDay`) that would trigger this specific case!